encrypt_util.go

```diff
@@ -74,21 +74,28 @@ func encryptStream(
 	chunk := make([]byte, chunkSize)
 
 	// encrypt file with CBC
+	padded := false
 	for {
 		n, err := src.Read(chunk)
 		if n == 0 || err != nil {
 			break
 		} else if n%aes.BlockSize != 0 || n != chunkSize {
 			chunk = padBytesLength(chunk[:n], aes.BlockSize)
+			padded = true
```

Comment:
The second condition in this `else if` block is a little confusing. Shouldn't the `n%aes.BlockSize` be enough?

Reply:
As chunk is allocated with 65536 (chunkSize) bytes, it will be fully filled with 65536 bytes no matter how many bytes src.Read(chunk) has read. The condition `n != chunkSize` in the `else if` block was needed to avoid getting unnecessary bytes from chunk.

To avoid confusion, I have updated the code and added more comments so it should be clear and consistent with the python driver's implementation. Also added some extra test cases.